neuro_extras/main.py

```diff
@@ -471,19 +435,100 @@ async def _nonstorage_cp(
         ]
     else:
         raise ValueError("Unknown cloud provider")
-    click.echo(f"Running {command} {' '.join(args)}")
+    click.echo(f"Running '{command} {' '.join(args)}'")
     subprocess = await asyncio.create_subprocess_exec(command, *args)
     returncode = await subprocess.wait()
+    if UrlType.get_type(source_url) == UrlType.LOCAL and remove_source:
+        _remove_local(source_url.path)
     if returncode != 0:
         raise click.ClickException("Cloud copy failed")
-    elif UrlType.get_type(source_url) == UrlType.LOCAL:
-        source_path = Path(source_url.path)
-        if remove_source:
-            if source_path.is_dir():
-                dir_util.remove_tree(str(source_path))
-            else:
-                if source_path.exists():
-                    source_path.unlink()
+
+
+async def _extract(src_url: URL, dst_url: URL, rm_src: bool) -> None:
+    src_path = Path(src_url.path)
+    dst_path = Path(dst_url.path)
+    if src_path.is_dir():
+        src_path = list(src_path.glob("*"))[0]
+    ext = ".".join(src_path.suffixes)
+    if ext in [".tar.gz", ".tgz"]:
+        command = "tar"
+        args = ["zxvf", str(src_path), "-C", str(dst_path)]
+    elif ext in [".tar.bz2", ".tbz2", ".tbz"]:
+        command = "tar"
+        args = ["jxvf", str(src_path), "-C", str(dst_url)]
+    elif ext == ".tar":
+        command = "tar"
+        args = ["xvf", str(src_path), "-C", str(dst_url)]
+    elif ext == ".gz":
+        command = "gunzip"
+        args = ["--keep", str(src_path), str(dst_url) + src_path.stem]
```

Comment:
here it's safe to use stem, it'll cut out only the high-level extension:
```
In [2]: p = Path("a.tar.gz")                                                                             

In [3]: p.stem                                                                                           
Out[3]: 'a.tar'
```

Should I add a comment here?

Reply:
ok, good idea, thx!

Reply:
reverted